AI/Nullkiller/Behaviors/StayAtTownBehavior.cpp

```diff
@@ -39,24 +39,15 @@ Goals::TGoalVec StayAtTownBehavior::decompose(const Nullkiller * ai) const
 
 	for(auto town : towns)
 	{
-		if(!town->hasBuilt(BuildingID::MAGES_GUILD_1))
-			continue;
-
 		ai->pathfinder->calculatePathInfo(paths, town->visitablePos());
 
 		for(auto & path : paths)
 		{
 			if(town->visitingHero && town->visitingHero.get() != path.targetHero)
 				continue;
 
-			if(!path.targetHero->hasSpellbook() || path.targetHero->mana >= 0.75f * path.targetHero->manaLimit())
```

Comment:
This was about mana regen using mages guild in town. Sure it may be worth to somehow estimate magic power in order to do this at all.

Reply:
I know and it will still be used for that. However, I also wanted a fallback for heros to go to towns when they don't have anything better to do. So I made it create tasks for that here for everything.